lib/orderbook/MatchingEngine.ts

```diff
@@ -106,21 +246,30 @@ class MatchingEngine {
         // there's no match with the best available maker order, so end the matching routine
         break;
       } else {
+        // get the order from the top of the queue, and remove its ref from the list as well
         const makerOrder = matchAgainst.poll()!;
+        const list = this.getOrderList(makerOrder);
+        list.delete(makerOrder.id);
```

Comment:
Maybe something for another PR, but I think it would be to `peek` the order at the top of the queue, and then only `poll()` and `list.delete()` once we confirm that it's getting filled entirely. That should be slightly more performant plus it feels like a cleaner approach.